rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -365,8 +400,14 @@ class topicnamesandtypesReaderListener : public ReaderListener
       tempMsg.length = change->serializedPayload.length;
       memcpy(tempMsg.buffer, change->serializedPayload.data, tempMsg.length);
       if (proxyData.readFromCDRMessage(&tempMsg)) {
+        auto partition_str = std::string("");
+        // don't use std::accumulate - schlemiel O(n2)
+        for (const auto & partition : proxyData.m_qos.m_partition.getNames()) {
```

Comment:
I implemented the same logic in `rmw_connext` given that the same code should be used for non-ros dds topic which may use multiple partitions. 
https://github.com/ros2/rmw_connext/blob/namespace_impl/rmw_connext_shared_cpp/src/shared_functions.cpp#L114-L117

The check for partition size 1 is implemented before that call:
https://github.com/ros2/rmw_fastrtps/pull/100/files#diff-8225b098f1ec4debcdcee03d65ae95d5R2779